lnwire/onion_error.go

```diff
@@ -1224,10 +1237,6 @@ func DecodeFailure(r io.Reader, pver uint32) (FailureMessage, error) {
 	if err := ReadElement(r, &failureLength); err != nil {
 		return nil, fmt.Errorf("unable to read error len: %v", err)
 	}
-	if failureLength > FailureMessageLength {
```

Comment:
I know that it is already checked in `lightning-onion`, but should we also error here if `length < FailureMessageLength`. It was agreed on the spec meeting to just enforce at least 256 lengths (since lnd already disallows anything other than 256).

Reply:
That check would also need to include the pad length, because just the failure message itself is allowed to be short. Not sure if we want to decode the pad length and verify the number of remaining bytes just for a sanity check?

Reply:
I think an extra check can't hurt to make sure that the padding is well-formed if the failure length is < 256

Reply:
Added the check, and now it turns out that the padding in the new longer test vector isn't correct. Will fix.

Reply:
Updated https://github.com/lightning/bolts/pull/1021 and test vector in this PR.